google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -625,6 +635,24 @@ public void onTransactionMetadata(Transaction transaction, boolean shouldInclude
       }
     }
 
+    /**
+     * In read-write transactions, the precommit token with the highest sequence number from this
+     * transaction attempt will be tracked and included in the
+     * [Commit][google.spanner.v1.Spanner.Commit] request for the transaction.
+     */
+    @Override
+    public void onPrecommitToken(MultiplexedSessionPrecommitToken token) {
+      if (token == null) return;
+      synchronized (precommitTokenLock) {
+        if (this.latestPrecommitToken == null
+            || token.getSeqNum() > this.latestPrecommitToken.getSeqNum()) {
+          this.latestPrecommitToken = token;
+          System.out.println("Updating precommit token to " + this.latestPrecommitToken);
+          txnLogger.log(Level.ALL, "Updating precommit token to " + this.latestPrecommitToken);
```

Comment:
I think that you want log level FINE for this.

Reply:
Yeah. Thanks for pointing this